internal/cloud/realm/dependencies.go

```diff
@@ -0,0 +1,87 @@
+package realm
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"mime"
+	"mime/multipart"
+	"net/http"
+	"os"
+
+	"github.com/10gen/realm-cli/internal/utils/api"
+)
+
+const (
+	dependenciesPathPattern        = appPathPattern + "/dependencies"
+	dependenciesArchivePathPattern = dependenciesPathPattern + "/archive"
+
+	paramFile = "file"
+)
+
+func (c *client) ImportDependencies(groupID, appID, uploadPath string) error {
+	file, fileErr := os.Open(uploadPath)
+	if fileErr != nil {
+		return fileErr
+	}
+	defer file.Close()
+
+	fileInfo, fileInfoErr := file.Stat()
+	if fileInfoErr != nil {
+		return fileInfoErr
+	}
+
+	body := &bytes.Buffer{}
+	w := multipart.NewWriter(body)
+
+	form, formErr := w.CreateFormFile(paramFile, fileInfo.Name())
+	if formErr != nil {
+		return formErr
+	}
+
+	if _, err := io.Copy(form, file); err != nil {
+		return err
+	}
+	if err := w.Close(); err != nil {
+		return err
+	}
+
+	res, resErr := c.do(
+		http.MethodPost,
+		fmt.Sprintf(dependenciesPathPattern, groupID, appID),
+		api.RequestOptions{
+			Body:        body,
+			ContentType: w.FormDataContentType(),
+		},
+	)
+	if resErr != nil {
+		return resErr
+	}
+	if res.StatusCode != http.StatusNoContent {
+		return api.ErrUnexpectedStatusCode{"import dependencies", res.StatusCode}
+	}
+	return nil
+}
+
+func (c *client) ExportDependencies(groupID, appID string) (string, io.ReadCloser, error) {
```

Comment:
Can we get a comment as to why we are exporting this as an `io.ReadCloser` instead of a struct? I see we simply use that to write to a file, but let's call that out here since other client calls return a struct. Or - I wouldn't mind returning a struct with FileName and Contents.

Reply:
so this is likely just my inexperience with low-level i/o stuff, but we do it this way in the old cli so i presumed that was an optimization to avoid having to "read" the contents twice?

if you have a suggested route, i'm all ears.  i imagine its something to do with:

```go
// please ignore the name...
type MyExportedDependencies struct {
  filename string
  contents []byte // is this then populated with the return value of ioutil.ReadAll ???
}
```